draft-ietf-mls-protocol.md

```diff
@@ -1013,7 +1052,6 @@ struct {
   PublicKey tree<1..2^32-1>;
   GroupOperation transcript<0..2^32-1>;
   opaque init_secret<0..255>;
```

Comment:
This leaks the group secret of the current epoch to the new participant.

Reply:
If we go for the solution proposed on the mailing list, this should be the hash of the current group secret and the other members should perform the hash as they process the Add message.

Reply:
You guys have an off-by-one error :)  This is the `init_secret` at the **output** of the current epoch's key schedule, not the input.  So it can't be used to derive the current epoch secret.

The whole reason that we have an `init_secret` (as opposed to chaining from one `epoch_secret` to the next) is so that you can give it to someone to allow them to initialize without giving away the current epoch secrets.  

@beurdouche not sure what you mean about going backward.  Key schedule is all HDKFing along, so it's not reversible.

Reply:
In any case, I'll add some notes to clarify.

Reply:
Well I understand how you read it but the diagram is ill-numbered then, because the `init_secret` is actually at the *input* of the the key schedule. The very first one is the `init_secret[0] = {0}` and is actually needed.

Reply:
I don't think you understand us regarding the Group operation transcript. If it is not a hash, it contains all group operation messages since the beginning of the group. You can trivially process all these and compute all keys for all previous application messages hence trivially defeating all secrecy.  It also works for future messages as long as the participant you compromise doesn't update. This needs to be a hash, I don't understand why you would think this is safe. @raphaelrobert ?

Reply:
Oh, I think it is safe because in order to process the GroupOperation messages, you need to have one of the corresponding private keys, which means you need to be in the group!

I'm happy to convert the transcript to a hash (there's no reason for anyone to have the whole thing, unlike the roster and the tree).  But I don't think it's necessary for this to be secure.

Reply:
cf #73